### PR TITLE
feat(designer): Adding support for warnings and custom messages to ErrorsPanel

### PR DESCRIPTION
**Pull request by wifun**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
    * Adds capability for host to supply custom errors/warnings to be displayed in ErrorsPanel
    * Adds new `setHostErrorMessages` for hosts to use to supply custom errors/messages

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
    * no breaking changes

- **Please Include Screenshots or Videos of the intended change**:
image

### CODE REVIEW

**Comment by wifun** [comment]
Still doing validation on my end, leaving this comment up to block merge

**Review by wifun — COMMENTED** [review]
test

File: libs/utils/src/lib/helpers/run.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { isObject } from '../helpers';
 import type { CallbackInfo, CallbackInfoWithRelativePath } from '../models';
+import { isObject } from './functions';
```
Comment (wifun) [inline libs/utils/src/lib/helpers/run.ts:2]: Will revert

File: libs/designer/src/lib/core/state/panel/panelInterfaces.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { PanelLocation } from '@microsoft/designer-ui';
 
-export type PanelMode = 'Operation' | 'Discovery' | 'WorkflowParameters' | 'NodeSearch' | 'Error' | 'Connection';
+export type PanelMode = 'Operation' | 'Discovery' | 'WorkflowParameters' | 'NodeSearch' | 'FlowChecker' | 'Connection';
```
Comment (wifun) [inline libs/designer/src/lib/core/state/panel/panelInterfaces.ts:3]: Still doing validation, adding comment to block merge

**Comment by hartra344** [comment]
why 2 PRs for the same thing? Is the other one valid?

File: libs/designer/src/lib/core/state/panel/panelInterfaces.ts (hunk 2 above)
Comment (hartra344) [reply libs/designer/src/lib/core/state/panel/panelInterfaces.ts:3]: Will this cause this to be a breaking change to the Hosts?

Comment (wifun) [reply libs/designer/src/lib/core/state/panel/panelInterfaces.ts:3]: yes should be minor, I've called it out in the PR description. For example for PAuto to open the panel we will have to specify the new name.

**Comment by wifun** [comment]
> why 2 PRs for the same thing? Is the other one valid?

Closed it, the other was targeting the version PAuto uses so I could verify it worked for our needs.

File: libs/designer/src/lib/core/state/panel/panelInterfaces.ts (hunk 2 above)
Comment (hartra344) [reply libs/designer/src/lib/core/state/panel/panelInterfaces.ts:3]: Why are we changing the underlying name then? We should be able to change the text without changing the names under,Error still makes plenty of sense.

Comment (wifun) [reply libs/designer/src/lib/core/state/panel/panelInterfaces.ts:3]: There are a couple components and classnames that use 'Error' which are now shared (ex. nodeErrorCard, nodeErrors, node-error-category-title, etc.).
I thought it be clearer to rename all references of 'ErrorsPanel' to FlowCheckerPanel and remaining references of 'error' to 'message'.

Not a strong preference, so if you would like to keep a mix such as keeping 'errorsPanel.tsx' as is and doing nodeErrors -> nodeMessages.tsx, node-error-category -> node-message-category I can do that

**Review by rllyy97 — APPROVED** [review]

**Comment by rllyy97** [comment]
Accidentally linked this PR on an unrelated pr

**Pull request merged**